src/sagas/tokens.js

```diff
@@ -281,6 +283,48 @@ export function* monitorSelectedToken() {
   }
 }
 
+/**
+ * This saga will create a channel to queue PROPOSAL_TOKEN_FETCH_REQUESTED actions and
+ * consumers that will run in parallel consuming those actions.
+ *
+ * More information about channels can be read on https://redux-saga.js.org/docs/api/#takechannel
+ */
+function* fetchProposalTokenDataQueue() {
+  const fetchProposalTokenDataChannel = yield call(channel);
+
+  // Fork CONCURRENT_FETCH_REQUESTS threads to download token metadata
+  for (let i = 0; i < CONCURRENT_FETCH_REQUESTS; i += 1) {
+    yield fork(fetchProposalTokenDataConsumer, fetchProposalTokenDataChannel);
+  }
+
+  while (true) {
+    const action = yield take(types.PROPOSAL_TOKEN_FETCH_REQUESTED);
+    yield put(fetchProposalTokenDataChannel, action);
+  }
+}
+
+/**
+ * This saga will consume the fetchProposalTokenDataChannel for PROPOSAL_TOKEN_FETCH_REQUESTED actions
+ * and wait until the PROPOSAL_TOKEN_FETCH_SUCCESS action is dispatched with the specific proposalId
+ */
+function* fetchProposalTokenDataConsumer(fetchTokenMetadataChannel) {
```

Comment:
This is not a token metadata channel, right? It's more like a token data

Reply:
On [ad3eace](https://github.com/HathorNetwork/hathor-wallet/pull/401/commits/ad3eace4c78441d95903f2e43d881ca63d70d740) adjusted jsdocs, variable names and error messages to clarify that this context is about token data ( name and symbol )